AutomatticTracks/src/main/java/com/automattic/android/tracks/CrashLogging/CrashLoggingDataProvider.kt

```diff
@@ -0,0 +1,59 @@
+package com.automattic.android.tracks.crashlogging
+
+import com.automattic.android.tracks.TracksUser
+import java.util.Locale
+
+interface CrashLoggingDataProvider {
+    /**
+     * Provides [CrashLogging] with the Sentry DSN for this application.
+     * @return The Sentry DSN for this application.
+     */
+    val sentryDSN: String
+
+    /**
+     * Provides [CrashLogging] with information on whether the user has opted out
+     * of data collection.
+     *
+     * The implementation of this method should fetch this value from persistent storage, and should
+     * never call `CrashLogging.getUserHasOptedOut()`.
+     *
+     * @return A value indicating whether or not the user has opted out of data collection.
+     */
+    val userHasOptedOut: Boolean
+
+    /**
+     * Provides [CrashLogging] with information on what type of build this is.
+     * @return The build type
+     */
+    val buildType: String
+
+    /**
+     * Provides [CrashLogging] with the name of this release.
+     * @return The release name
+     */
+    val releaseName: String
+
+    /**
+     * Provides [CrashLogging] with information about the current user.
+     * @return A `TracksUser` object containing data about the current user.
+     *
+     * @see TracksUser
+     */
+    val currentUser: TracksUser?
+
+    /**
+     * Provides the [CrashLogging] with information about the current application state.
+     */
+    val applicationContext: Map<String, String?>
```

Comment:
I don't know. This was the api that was used before, I don't have enough knowledge to decide on any change here.

If I was about to guess, I'd say that there might be cases when some value stored as `aplicationContext` _might_ be `null` because why not? Information that something _does not exist_ is still valuable information.

Reply:
I've ended changing the signature of `applicationContext` and `userContext` to `Map<String, String>`. The reason for that is that `Sentry SDK` won't accept a `Map` with a null value for a map's entry. More details and fix: https://github.com/getsentry/sentry-java/issues/1366

We could leave this interface and map `null` to an empty string or maybe drop all entries that have null value but I'd like to leave this decision to the library consumer.